Deprecated v8 references

Here are the errors I've encountered:

```
npm ERR! In file included from ../routines/level1/dot.cc:2:
npm ERR! ../routines/level1/dot.cc: In function ‘void ddot(const v8::FunctionCallbackInfo<v8::Value>&)’:
npm ERR! ../lib/routines.h:7:46: error: ‘class v8::ArrayBuffer’ has no member named ‘GetContents’
npm ERR!     7 | (static_cast<unsigned char*>(view->Buffer()->GetContents().Data()) + view->ByteOffset())
npm ERR!       |                                              ^~~~~~~~~~~
npm ERR! ../routines/level1/dot.cc:6:53: note: in expansion of macro ‘GET_CONTENTS’
npm ERR!     6 |         const double *x = reinterpret_cast<double*>(GET_CONTENTS(info[1].As<v8::Float64Array>()));
```
... and many more like that.

Also:

```
npm ERR! In file included from ../src/index.cc:1:
npm ERR! /home/larry/.cache/node-gyp/19.4.0/include/node/node.h:993:7: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>)’ to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
npm ERR!   993 |       (node::addon_register_func) (regfunc),                          \
npm ERR!       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
npm ERR! /home/larry/.cache/node-gyp/19.4.0/include/node/node.h:1027:3: note: in expansion of macro ‘NODE_MODULE_X’
npm ERR!  1027 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
npm ERR!       |   ^~~~~~~~~~~~~
npm ERR! ../src/index.cc:121:1: note: in expansion of macro ‘NODE_MODULE’
npm ERR!   121 | NODE_MODULE(addon, Init)
npm ERR!       | ^~~~~~~~~~~
```

Currently, the npm package refers to `GetContents`, but my compiler is very eager to tell me `‘class v8::ArrayBuffer’ has no member named ‘GetContents’`. As mentioned in [this SO post](https://stackoverflow.com/questions/69290893/getcontents-in-v8arraybuffer), this has been replaced with the [`GetBackingStore` function](https://v8.github.io/api/head/classv8_1_1ArrayBuffer.html#aaa29e281f94990cbf0a5484140fffb50).

Additionally, it looks like `node::addon_register_func` has had its function signature changed, and now requires two additional parameters.

Everything I know about bindings I learned in the last fifteen minutes trying to gather enough information for this issue to be intelligible.

--- a/lib/routines.h
+++ b/lib/routines.h
@@ -3,8 +3,13 @@
 
 #include <node.h>
 
-#define GET_CONTENTS(view) \
-(static_cast<unsigned char*>(view->Buffer()->GetContents().Data()) + view->ByteOffset())
+#if V8_MAJOR_VERSION < 8
+  #define GET_CONTENTS(view) \
+  (static_cast<unsigned char*>(view->Buffer()->GetContents().Data()) + view->ByteOffset())
+#else
+  #define GET_CONTENTS(view) \
+  (static_cast<unsigned char*>(view->Buffer()->GetBackingStore()->Data()) + view->ByteOffset())
+#endif
 
 void dasum(const v8::FunctionCallbackInfo<v8::Value>& info);
 void sasum(const v8::FunctionCallbackInfo<v8::Value>& info);

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "nblas",
-  "version": "2.1.9",
+  "version": "2.1.10",
   "description": "C++ bindings for all single- and double-precision CBLAS (Basic Linear Algebra Subprograms) routines.",
   "main": "built/index.js",
   "types": "built/index.d.ts",

--- a/routines/level2/gbmv.cc
+++ b/routines/level2/gbmv.cc
@@ -13,7 +13,7 @@ void dgbmv(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const double *x = reinterpret_cast<double*>(GET_CONTENTS(info[8].As<v8::Float64Array>()));
 	const int inc_x = info[9].As<v8::Integer>()->Value();
 	const double beta = info[10].As<v8::Number>()->Value();
-	double *y = reinterpret_cast<double*>(info[11].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	double *y = reinterpret_cast<double*>(GET_CONTENTS(info[11].As<v8::Float64Array>()));
 	const int inc_y = info[12].As<v8::Integer>()->Value();
 	cblas_dgbmv(CblasRowMajor, trans, m, n, kl, ku, alpha, a, lda, x, inc_x, beta, y, inc_y);
 }
@@ -30,7 +30,7 @@ void sgbmv(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const float *x = reinterpret_cast<float*>(GET_CONTENTS(info[8].As<v8::Float32Array>()));
 	const int inc_x = info[9].As<v8::Integer>()->Value();
 	const float beta = info[10].As<v8::Number>()->Value();
-	float *y = reinterpret_cast<float*>(info[11].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	float *y = reinterpret_cast<float*>(GET_CONTENTS(info[11].As<v8::Float32Array>()));
 	const int inc_y = info[12].As<v8::Integer>()->Value();
 	cblas_sgbmv(CblasRowMajor, trans, m, n, kl, ku, alpha, a, lda, x, inc_x, beta, y, inc_y);
 }
@@ -46,8 +46,8 @@ void zgbmv(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[7].As<v8::Integer>()->Value();
 	const double *x = reinterpret_cast<double*>(GET_CONTENTS(info[8].As<v8::Float64Array>()));
 	const int inc_x = info[9].As<v8::Integer>()->Value();
-	const double *beta = reinterpret_cast<double*>(info[10].As<v8::Float64Array>()->Buffer()->GetContents().Data());
-	double *y = reinterpret_cast<double*>(info[11].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	const double *beta = reinterpret_cast<double*>(GET_CONTENTS(info[10].As<v8::Float64Array>()));
+	double *y = reinterpret_cast<double*>(GET_CONTENTS(info[11].As<v8::Float64Array>()));
 	const int inc_y = info[12].As<v8::Integer>()->Value();
 	cblas_zgbmv(CblasRowMajor, trans, m, n, kl, ku, alpha, a, lda, x, inc_x, beta, y, inc_y);
 }
@@ -63,8 +63,8 @@ void cgbmv(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[7].As<v8::Integer>()->Value();
 	const float *x = reinterpret_cast<float*>(GET_CONTENTS(info[8].As<v8::Float32Array>()));
 	const int inc_x = info[9].As<v8::Integer>()->Value();
-	const float *beta = reinterpret_cast<float*>(info[10].As<v8::Float32Array>()->Buffer()->GetContents().Data());
-	float *y = reinterpret_cast<float*>(info[11].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	const float *beta = reinterpret_cast<float*>(GET_CONTENTS(info[10].As<v8::Float32Array>()));
+	float *y = reinterpret_cast<float*>(GET_CONTENTS(info[11].As<v8::Float32Array>()));
 	const int inc_y = info[12].As<v8::Integer>()->Value();
 	cblas_cgbmv(CblasRowMajor, trans, m, n, kl, ku, alpha, a, lda, x, inc_x, beta, y, inc_y);
 }

--- a/routines/level3/gemm.cc
+++ b/routines/level3/gemm.cc
@@ -13,7 +13,7 @@ void dgemm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const double *b = reinterpret_cast<double*>(GET_CONTENTS(info[8].As<v8::Float64Array>()));
 	const int ldb = info[9].As<v8::Integer>()->Value();
 	const double beta = info[10].As<v8::Number>()->Value();
-	double *c = reinterpret_cast<double*>(info[11].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	double *c = reinterpret_cast<double*>(GET_CONTENTS(info[11].As<v8::Float64Array>()));
 	const int ldc = info[12].As<v8::Integer>()->Value();
 	cblas_dgemm(CblasRowMajor, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -30,7 +30,7 @@ void sgemm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const float *b = reinterpret_cast<float*>(GET_CONTENTS(info[8].As<v8::Float32Array>()));
 	const int ldb = info[9].As<v8::Integer>()->Value();
 	const float beta = info[10].As<v8::Number>()->Value();
-	float *c = reinterpret_cast<float*>(info[11].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	float *c = reinterpret_cast<float*>(GET_CONTENTS(info[11].As<v8::Float32Array>()));
 	const int ldc = info[12].As<v8::Integer>()->Value();
 	cblas_sgemm(CblasRowMajor, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -46,8 +46,8 @@ void zgemm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[7].As<v8::Integer>()->Value();
 	const double *b = reinterpret_cast<double*>(GET_CONTENTS(info[8].As<v8::Float64Array>()));
 	const int ldb = info[9].As<v8::Integer>()->Value();
-	const double *beta = reinterpret_cast<double*>(info[10].As<v8::Float64Array>()->Buffer()->GetContents().Data());
-	double *c = reinterpret_cast<double*>(info[11].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	const double *beta = reinterpret_cast<double*>(GET_CONTENTS(info[10].As<v8::Float64Array>()));
+	double *c = reinterpret_cast<double*>(GET_CONTENTS(info[11].As<v8::Float64Array>()));
 	const int ldc = info[12].As<v8::Integer>()->Value();
 	cblas_zgemm(CblasRowMajor, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -63,8 +63,8 @@ void cgemm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[7].As<v8::Integer>()->Value();
 	const float *b = reinterpret_cast<float*>(GET_CONTENTS(info[8].As<v8::Float32Array>()));
 	const int ldb = info[9].As<v8::Integer>()->Value();
-	const float *beta = reinterpret_cast<float*>(info[10].As<v8::Float32Array>()->Buffer()->GetContents().Data());
-	float *c = reinterpret_cast<float*>(info[11].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	const float *beta = reinterpret_cast<float*>(GET_CONTENTS(info[10].As<v8::Float32Array>()));
+	float *c = reinterpret_cast<float*>(GET_CONTENTS(info[11].As<v8::Float32Array>()));
 	const int ldc = info[12].As<v8::Integer>()->Value();
 	cblas_cgemm(CblasRowMajor, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }

--- a/routines/level3/symm.cc
+++ b/routines/level3/symm.cc
@@ -12,7 +12,7 @@ void dsymm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const double *b = reinterpret_cast<double*>(GET_CONTENTS(info[7].As<v8::Float64Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
 	const double beta = info[9].As<v8::Number>()->Value();
-	double *c = reinterpret_cast<double*>(info[10].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	double *c = reinterpret_cast<double*>(GET_CONTENTS(info[10].As<v8::Float64Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_dsymm(CblasRowMajor, side, uplo, m, n, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -28,7 +28,7 @@ void ssymm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const float *b = reinterpret_cast<float*>(GET_CONTENTS(info[7].As<v8::Float32Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
 	const float beta = info[9].As<v8::Number>()->Value();
-	float *c = reinterpret_cast<float*>(info[10].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	float *c = reinterpret_cast<float*>(GET_CONTENTS(info[10].As<v8::Float32Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_ssymm(CblasRowMajor, side, uplo, m, n, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -43,8 +43,8 @@ void zsymm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[6].As<v8::Integer>()->Value();
 	const double *b = reinterpret_cast<double*>(GET_CONTENTS(info[7].As<v8::Float64Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
-	const double *beta = reinterpret_cast<double*>(info[9].As<v8::Float64Array>()->Buffer()->GetContents().Data());
-	double *c = reinterpret_cast<double*>(info[10].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	const double *beta = reinterpret_cast<double*>(GET_CONTENTS(info[9].As<v8::Float64Array>()));
+	double *c = reinterpret_cast<double*>(GET_CONTENTS(info[10].As<v8::Float64Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_zsymm(CblasRowMajor, side, uplo, m, n, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -59,8 +59,8 @@ void csymm(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[6].As<v8::Integer>()->Value();
 	const float *b = reinterpret_cast<float*>(GET_CONTENTS(info[7].As<v8::Float32Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
-	const float *beta = reinterpret_cast<float*>(info[9].As<v8::Float32Array>()->Buffer()->GetContents().Data());
-	float *c = reinterpret_cast<float*>(info[10].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	const float *beta = reinterpret_cast<float*>(GET_CONTENTS(info[9].As<v8::Float32Array>()));
+	float *c = reinterpret_cast<float*>(GET_CONTENTS(info[10].As<v8::Float32Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_csymm(CblasRowMajor, side, uplo, m, n, alpha, a, lda, b, ldb, beta, c, ldc);
 }

--- a/routines/level3/syr2k.cc
+++ b/routines/level3/syr2k.cc
@@ -12,7 +12,7 @@ void dsyr2k(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const double *b = reinterpret_cast<double*>(GET_CONTENTS(info[7].As<v8::Float64Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
 	const double beta = info[9].As<v8::Number>()->Value();
-	double *c = reinterpret_cast<double*>(info[10].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	double *c = reinterpret_cast<double*>(GET_CONTENTS(info[10].As<v8::Float64Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_dsyr2k(CblasRowMajor, uplo, trans, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -28,7 +28,7 @@ void ssyr2k(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const float *b = reinterpret_cast<float*>(GET_CONTENTS(info[7].As<v8::Float32Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
 	const float beta = info[9].As<v8::Number>()->Value();
-	float *c = reinterpret_cast<float*>(info[10].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	float *c = reinterpret_cast<float*>(GET_CONTENTS(info[10].As<v8::Float32Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_ssyr2k(CblasRowMajor, uplo, trans, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -43,8 +43,8 @@ void zsyr2k(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[6].As<v8::Integer>()->Value();
 	const double *b = reinterpret_cast<double*>(GET_CONTENTS(info[7].As<v8::Float64Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
-	const double *beta = reinterpret_cast<double*>(info[9].As<v8::Float64Array>()->Buffer()->GetContents().Data());
-	double *c = reinterpret_cast<double*>(info[10].As<v8::Float64Array>()->Buffer()->GetContents().Data());
+	const double *beta = reinterpret_cast<double*>(GET_CONTENTS(info[9].As<v8::Float64Array>()));
+	double *c = reinterpret_cast<double*>(GET_CONTENTS(info[10].As<v8::Float64Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_zsyr2k(CblasRowMajor, uplo, trans, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
@@ -59,8 +59,8 @@ void csyr2k(const v8::FunctionCallbackInfo<v8::Value>& info) {
 	const int lda = info[6].As<v8::Integer>()->Value();
 	const float *b = reinterpret_cast<float*>(GET_CONTENTS(info[7].As<v8::Float32Array>()));
 	const int ldb = info[8].As<v8::Integer>()->Value();
-	const float *beta = reinterpret_cast<float*>(info[9].As<v8::Float32Array>()->Buffer()->GetContents().Data());
-	float *c = reinterpret_cast<float*>(info[10].As<v8::Float32Array>()->Buffer()->GetContents().Data());
+	const float *beta = reinterpret_cast<float*>(GET_CONTENTS(info[9].As<v8::Float32Array>()));
+	float *c = reinterpret_cast<float*>(GET_CONTENTS(info[10].As<v8::Float32Array>()));
 	const int ldc = info[11].As<v8::Integer>()->Value();
 	cblas_csyr2k(CblasRowMajor, uplo, trans, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }

